### ISSUE TITLE
data(), key(), enter(), exit() 

### ISSUE THREAD

**Issue by sttawm** [mapsense/mapsense.js#22]
I've been thinking a bit about the heat-map (AKA gridded binning) problem and the steps necessary to enable heat-map updates in response to filtering. Generating contour lines also come to mind (essentially, any visualization of derived data does).

I've been playing with the idea of adding a few functions:

```js
/**
  * Derive new features from the fetched and cached features, re-bind them with the feature selections, and trigger all subsequent processing.
  */ 
topojson.data(function([features]) { });

/**
 * Defines the d3 key function to be used with .data()
 */
topojson.keys(function(feature) { return feature.id });

/**
  * Set the enter method to be invoked for each enter-selection.
  */
topojson.enter(function(enterSelection) { });

/**
  * Set the exit method to be invoked for each exit-selection.
  */
topojson.exit(function(exitSelection) { })

/**
 * Set the update method to be invoked for each update-selection, and immediately invoked it on the current selection.
 */ 
topojson.update(function(updateSelection) { })
```

It'd probably also make sense to allow these functions to optionally take the tile as an argument.

=====================

Note that, today, selection(...) would be equivalent (an alias?) of update(...). We may prefer a sort of "preUpdate" selection (to gain access to those elements that were bound to data both before and after .data())

This would allow us, in response to a new filter, to do something like:

```js
var filter = function(f) { return f.properties.population > 5; };
topojson.data(function(features) {
   return binCounts(features.filter(filter)); // returns polygons
});
```

======================

To get a sense of what it might look like, and to better explain myself, I have an (incomplete) [commit](https://github.com/mapsense/mapsense.js/commit/055ed156aa8dc0ebdd59d3e296882bd0fb9d0943).
Basically, before digging into how to re-trigger the "draw()" function I'd like to make sure this is something we actually want.
